File: nysol_python/mcmd/methods/convTex/db/smp/mfldname.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mfldname'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,itemID,10月
a,xx,11
b,yy,122
c,zz,
'''
db['idatas'].append(data)

data={}
data['name']='dat2.csv'
data['text']='''
a,xx,11
b,yy,122
c,zz,
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
項目名の ``customer`` を「cust」に、「10月」を「Oct.」に変更する。
'''
script['sh_code']='''
mfldname f=顧客:cust,10月:Oct. i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mfldname(f="customer:cust,10月:Oct.", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='項目名変更'
script['comment']='''
項目名を ``x,y,z`` に変更する。
'''
script['sh_code']='''
mfldname n=x,y,z i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mfldname(n="x,y,z", i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='項目名行がないデータ'
script['comment']='''
'''
script['sh_code']='''
mfldname -nfni n=x,y,z i=dat2.csv o=rsl3.csv
'''
script['py_code']='''
nm.mfldname(nfni=True, n="x,y,z", i="dat2.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mfldname.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
customer,itemID,october
a,xx,11
b,yy,122
c,zz,
EOF
)}

File.open("dat2.csv","w"){|fpw| fpw.write(
<<'EOF'
a,xx,11
b,yy,122
c,zz,
EOF
)}


File.open("dat3.csv","w"){|fpw| fpw.write(
<<'EOF'
customer%r,itemID,october
c,zz,
b,yy,122
a,xx,11
EOF
)}


############## Example 1
title="Basic Example"
comment=<<'EOF'
Change column name from “customer” to “cust”, and “october” to “oct”.
EOF
scp=<<'EOF'
more dat1.csv
mfldname f=customer:cust,october:oct. i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Rename column"
comment=<<'EOF'
Change field names to \verb|x,y,z|.
EOF
scp=<<'EOF'
mfldname n=x,y,z i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Data without field names"
comment=<<'EOF'
EOF
scp=<<'EOF'
more dat2.csv
mfldname -nfni n=x,y,z i=dat2.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

############## Example 4
title="Remove sort order symbols in field names"
comment=<<'EOF'
EOF
scp=<<'EOF'
more dat3.csv
mfldname -q  i=dat3.csv o=rsl4.csv
more rsl4.csv
EOF
run(scp,title,comment)
